File: Python/1021.py
# 1021. Remove Outermost Parentheses
# https://leetcode.com/problems/remove-outermost-parentheses/

class Solution:
  def removeOuterParentheses(self, s: str) -> str:
    cnt = 0
    out = ""

    for c in s:
      if (c == '('):
        cnt += 1
      else:
        cnt -= 1

      if ((c == '(' and cnt == 1) or (c == ')' and cnt == 0)):
        continue

      out += c

    return out

File: Rust/1021.rs
/**
 * 1021. Remove Outermost Parentheses
 * https://leetcode.com/problems/remove-outermost-parentheses/
 */

impl Solution {
  pub fn remove_outer_parentheses(s: String) -> String {
    let mut cnt: i32 = 0;
    let mut out: String = "".to_string();

    for c in s.chars() {
      if c == '(' {
        cnt += 1;
      } else {
        cnt -= 1;
      }

      if (c == '(' && cnt == 1) || (c == ')' && cnt == 0) {
        continue;
      }

      out.push(c);
    }

    return out;
  }
}
